anchore_engine/services/policy_engine/engine/vulns/stores.py

```diff
@@ -169,62 +194,57 @@ def _delete_entry(self, entry):
                     break
                 except:
                     log.exception(
-                        "Could not delete vulnerabilities report from cache, will retry. Bucket={}, Key={}".format(
+                        "Could not delete vulnerabilities report from store, will retry. Bucket={}, Key={}".format(
                             bucket, key
                         )
                     )
                     retry -= 1
             else:
                 log.error(
-                    "Could not flush vulnerabilities report from cache after all retries, may be orphaned. Will remove from index."
+                    "Could not flush vulnerabilities report from store after all retries, may be orphaned. Will remove from index."
                 )
 
         session.delete(entry)
         session.flush()
 
-    def flush(self):
+    def delete_all(self):
         """
-        Flush all cache entries for the given image
+        Flush all report entries for the given image
         :return:
         """
         session = get_session()
-        for entry in session.query(CachedVulnerabilities).filter_by(
+        for entry in session.query(DbImageVulnerabilities).filter_by(
             account_id=self.image.user_id, image_digest=self.image.digest
         ):
             try:
                 self._delete_entry(entry)
```

Comment:
Is this still necessary? Looks like a no-op

Reply:
How's it a no-op? It still deletes the existing record before creating a new one in the save() workflow